metricflow/test/table_snapshot/table_snapshots.py

```diff
@@ -0,0 +1,233 @@
+            for table_snapshot in table_snapshots:
+                if table_snapshot.table_name in self._table_snapshots:
+                    raise ValueError(
+                        f"Table with duplicate name found: {table_snapshot.table_name}. "
+                        f"Conflicting paths: {table_snapshot.file_path} "
+                        f"{self._table_snapshots[table_snapshot.table_name].file_path}"
+                    )
+                self._table_snapshots[table_snapshot.table_name] = table_snapshot
+
+    @staticmethod
+    def _parse_config_yaml(file_path: Path) -> Sequence[SqlTableSnapshot]:
+        """Parse the YAML file at the given path into table snapshots."""
+        results = []
+        document_key = "table_snapshot"
+        with open(file_path) as f:
+            file_contents = f.read()
+            for config_document in yaml.load_all(stream=file_contents, Loader=yaml.SafeLoader):
+                # The config document can be None if there is nothing but white space between two `---`
+                # this isn't really an issue, so lets just swallow it
+                if config_document is None:
+                    continue
+                if not isinstance(config_document, dict):
+                    raise TableSnapshotParseException(
+                        f"Table snapshot YAML must be a dict. Got `{type(config_document)}`: {config_document}"
+                    )
+
+                keys = tuple(x for x in config_document.keys())
+                if len(keys) != 1:
+                    raise TableSnapshotParseException(
+                        f"Table snapshot document should have one type of key, but this has {len(keys)}. "
+                        f"Found keys: {keys} in {file_path}",
+                    )
+
+                # retrieve last top-level key as type
+                document_type = next(iter(config_document.keys()))
+                object_cfg = config_document[document_type]
+                if document_type == document_key:
+                    try:
+                        results.append(SqlTableSnapshot(**object_cfg, file_path=Path(file_path)))
+                    except Exception as e:
+                        logger.exception(f"Error while parsing: {file_path}")
+                        raise TableSnapshotParseException(f"Error while parsing: {file_path}") from e
+                else:
+                    raise TableSnapshotParseException(f"Expected {document_key}, but got {document_type}")
+        return results
+
+    @staticmethod
+    def _find_all_yaml_file_paths(directory: Path) -> Sequence[Path]:  # noqa: D
+        """Recursively search through the given directory for YAML files."""
+        yaml_file_paths = []
+
+        for root, dirs, files in os.walk(directory):
+            for file in files:
+                if file.endswith(".yaml"):
+                    yaml_file_paths.append(Path(root, file))
+
+        return sorted(yaml_file_paths)
```

Comment:
I feel like we have a bunch of these, but we'll get rid of some of them soon so I guess it's ok.

Reply:
Yeah, was thinking of a loader class with generic typing, but put that on the back burner.